### DISCUSSION TITLE
Allow miner/worker to restart sealing using the cache rather than wiping it and starting from scratch

### DISCUSSION THREAD

**Discussion by askender** [filecoin-project/lotus#6782] [Ideas] [answered]
```
stores/local.go:391 /*/cache/s-t0**-**
```
if miner/worker quit and restart, cache will all be removed and redo precommit1. It waste a lot of time. Does it can be improved?

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment]
i think it has to be like that

#antidisestablishatarianism

**Comment by jennijuju** [comment]
@askender How do you think it can be improved?

**Reply by lanzafame** [reply]
When restarting, the miner should detect that the cached sector is present and continue from that point instead of wiping the cache sector and starting PC1 all over again. I am sure there are some reasons why this isn't currently done, but it would be nice to see this fixed.

**Comment by SgtCoin** [comment, marked as answer]
Archiving for future reference due to age.
